### PR TITLE
Display Game title and Title ID on Window title

### PR DESCRIPTION
**Pull request by bentokun**
- Add a SFO parser
- Add title_id and game_title variable to host state

### CODE REVIEW

File: src/emulator/host/include/host/state.h [hunk 1]
```diff
@@ -42,6 +42,8 @@ struct DisplayState {
 };
 
 struct HostState {
+    std::string title_id;
```
Comment (VelocityRa) [inline src/emulator/host/include/host/state.h]: Maybe add anything sfo related in a separate `SfoState` struct?

File: src/emulator/CMakeLists.txt [hunk 2]
```diff
@@ -51,6 +51,8 @@ relocation.cpp
 relocation.h
 vpk.cpp
 vpk.h
+sfo.h
```
Comment (petmac) [inline src/emulator/CMakeLists.txt]: So far I've been trying to keep file lists sorted, for a couple of reasons:
- Any ordering resists merge conflicts (and bad merges) better than no ordering
- Easier to compare to a directory listing

File: src/emulator/vpk.h [hunk 3]
```diff
@@ -24,4 +24,4 @@ struct MemState;
 template <class T>
 class Ptr;
 
-bool load_vpk(Ptr<const void> &entry_point, IOState &io, MemState &mem, const std::wstring& path);
+bool load_vpk(Ptr<const void> &entry_point, IOState &io, MemState &mem, const std::wstring& path, std::string& game_title, std::string& title_id);
```
Comment (petmac) [inline src/emulator/vpk.h]: I normally follow C standard library convention and put output parameters on the left.

File: src/emulator/sfo.h [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    struct SfoEntry {
+        SfoIndexTableEntry entry;
+        std::pair<std::string, std::string> data;
+    };
+
+    std::vector<SfoEntry> entries;
+};
+
+bool load_sfo(SfoFile &file, const std::vector<uint8_t>& data);
+bool load_sfo(SfoFile &file, const std::string path);
```
Comment (petmac) [inline src/emulator/sfo.h]: Is this function used?

Comment (bentokun) [reply src/emulator/sfo.h]: Will clean it, left over of that vfs pr.

**Pull request merged**